keep/common/fedora.py

```diff
@@ -319,8 +319,12 @@ def save(self, logMessage=None):
 
                 raise DuplicateContent(msg, pids, pid_cmodels)
 
-        return super(DigitalObject, self).save(logMessage)
+        # check if the mods content title of the object has been changed
+        # if changed, apply the change to the object in pidman
+        if self.mods.isModified(): # if title is changed it'd be reflected in mods
+            pidman.update_ark(noid=self.noid, name=self.mods.content.title)
```

Comment:
Makes sense and it's a nice catch as the labels (before and after) could be the same. Added logic to verify that the modification does happen in the label and also broke out the update method for future reference.